test/system/271-tcp-cors-server.bats

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Tests podman system service under systemd socket activation
+#
+
+load helpers
+load helpers.systemd
+
+SERVICE_NAME="podman_test_$(random_string)"
+
+SERVICE_TCP_ADDR="localhost:3113"
+
+SERVICE_FILE="$UNIT_DIR/$SERVICE_NAME.service"
+SOCKET_FILE="$UNIT_DIR/$SERVICE_NAME.socket"
+
+
+function setup() {
+    skip_if_remote "systemd tests are meaningless over remote"
+
+    basic_setup
+
+    cat > $SERVICE_FILE <<EOF
+[Unit]
+Description=Podman API Service
+Requires=podman.socket
+After=podman.socket
+Documentation=man:podman-system-service(1)
+StartLimitIntervalSec=0
+
+[Service]
+Type=exec
+KillMode=process
+Environment=LOGGING="--log-level=info"
+ExecStart=$PODMAN $LOGGING system service tcp:$SERVICE_TCP_ADDR --cors=* -t 10
+EOF
+    cat > $SOCKET_FILE <<EOF
+[Unit]
+Description=Podman API Socket
+Documentation=man:podman-system-service(1)
+
+[Socket]
+ListenStream=%t/podman/podman.sock
+SocketMode=0660
+
+[Install]
+WantedBy=sockets.target
+EOF
+
+    # ensure pause die before each test runs
+    if is_rootless; then
+        local pause_pid="$XDG_RUNTIME_DIR/libpod/tmp/pause.pid"
+        if [ -f $pause_pid ]; then
+            kill -9 $(cat $pause_pid) 2> /dev/null
+            rm -f $pause_pid
+        fi
+    fi
+    systemctl start "$SERVICE_NAME.socket"
+}
+
+function teardown() {
+    systemctl stop "$SERVICE_NAME.socket"
+    rm -f "$SERVICE_FILE" "$SOCKET_FILE"
+    systemctl daemon-reload
+    basic_teardown
+}
+
+@test "podman system service - tcp CORS " {
+    result=$(curl -s --max-time 10 -vvv $SERVICE_TCP_ADDR/_ping)
+    is "$result" ".*Access-Control.*" "Missing CORS headers"
+}
+
+@test "podman system service - CORS enabled in logs" {
+    result=$(run_podman system service --log-level="debug" --cors=* -t 1 | grep "CORS Headers were set to *")
```

Comment:
This does not do what you think it does. You probably want something like this instead:
```bash
    run_podman system service ...
    is "$output" ".*CORS Headers were set to.*" "debug log confirms CORS headers set"
```